### PR TITLE
Added encoding name translation

### PR DESCRIPTION
**Pull request by weirdan**
refs ticket AtomLinter/linter-phpcs#232

Maintenance notes:
 * `iconv-lite-aliases.json` is generated from
   `require('iconv-lite').encodings`
 * `iconv-lite-to-libiconv.json` is compiled manually by cross-referencing
   the aliases list with libiconv source code

### CODE REVIEW

**Comment by Arcanemagus** [comment]
* Can `iconv-lite-aliases.json` instead be done by depending on `iconv-lite`? I'd like to remove as much manual work from maintaining this as possible.
* Does case really matter in `iconv-lite-to-libiconv`? ([example](https://github.com/AtomLinter/linter-phpcs/pull/233/files#diff-476c5b211f59ac6f661f0b40d702a656R25))
* For encodings that aren't being translated, it doesn't seem there is a need to list them in `iconv-lite-to-libiconv`, since a `null` in there leads to the original being returned anyway.

**Comment by weirdan** [comment]
> Can iconv-lite-aliases.json instead be done by depending on iconv-lite? I'd like to remove as much manual work from maintaining this as possible.

Makes sense. What version should I depend on?

> Does case really matter in iconv-lite-to-libiconv? 

No idea. It may, I guess. If some implementations fail when they see `utf8` instead of `utf-8` I wouldn't be at all surprised if some other failed because of a case mismatch.

> For encodings that aren't being translated, it doesn't seem there is a need to list them in iconv-lite-to-libiconv, since a null in there leads to the original being returned anyway.

`null`s indicate encodings that are missing from libiconv. Originally I thought I would throw errors in this case, but if it's going to fail it may as well try to use the encoding, as phpcs may still accept it. I think instead of removing nulls it would be better to add a `console.warn()`. This way should a bug come in you would know it may be caused by potentially unsupported/unknown encoding.

**Comment by Arcanemagus** [comment]
> Makes sense. What version should I depend on?

I would do a caret dependency on the latest version, that way Greenkeeper will keep it up to date for us.

File: lib/main.js [hunk 1]
```diff
@@ -25,6 +28,19 @@ let showSource;
 let disableExecuteTimeout;
 let excludedSniffs;
 
+const translateEncodingName = (name) => {
+  const canonical = iconvLiteAliases[name] || name;
+  let ret;
+  if (iconvTranslation[canonical]) {
+    ret = iconvTranslation[canonical];
```
Comment (Arcanemagus) [inline lib/main.js]: You could just return directly here, then there is no need for the temporary variable.

Comment (weirdan) [inline lib/main.js]: You'd think so :laughing:, but:
```
> npm run-script lint

> linter-phpcs@1.5.6 lint /home/weirdan/src/linter-phpcs
> eslint .

/home/weirdan/src/linter-phpcs/lib/main.js
  35:10  error    Unnecessary 'else' after 'return'  no-else-return
```

Comment (Arcanemagus) [inline lib/main.js]: That's exactly what I mean 😛

```js
const translateEncodingName = (name) => {
  const canonical = iconvLiteAliases[name] || name;
  if (iconvTranslation[canonical]) {
    return iconvTranslation[canonical];
  }
  // eslint-disable-next-line no-console
  console.warn(`linter-phpcs:: Potentially unsupported encoding '${canonical}', but trying anyway`);
  return canonical;
};
```

Comment (weirdan) [inline lib/main.js]: Ah, **that's** what it complains about

Comment (weirdan) [inline lib/main.js]: It was like
```js
if (translation[canonical]) {
  return translation[canonical];
} else {
  console.warn('...');
  return canonical;
}
```

Comment (Arcanemagus) [inline lib/main.js]: Yea, I remember the first time I hit that I was rather confused until I realized what it meant.

**Comment by Arcanemagus** [comment]
Just waiting on CI for merging.

**Comment by weirdan** [comment]
@Arcanemagus , CI is done.

**Comment by Arcanemagus** [comment]
Published in v1.5.7. 🎉

**Pull request merged**